src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -866,29 +866,36 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
   }
 
   def cudaTranspose[N:Numeric:Manifest](implicit __pos: SourceContext) = cudaGlobalFun {
-    (in: Rep[Array[N]], out: Rep[Array[N]]) =>
+    (in: Rep[Array[N]], out: Rep[Array[N]], n: Rep[Int], m: Rep[Int]) =>
```

Comment:
@feiwang3311 Should I add tileDim and blockRows as parameters to the kernel? The calling function and the kernel need to agree on these constants for it to work correctly, so it might reduce errors if they were parameters instead of assumed.

Reply:
That is reasonable. But i think asking the caller to provide `tileDim` and `blockRows` is risky too, because the caller might use unreasonable `tileDim` or `blockRows`? I think maybe the comments of this function should state clearly what is `tileDim` and `blockRows` (since they are constant for kernel writer) and how it should be used by the caller.

Reply:
I think that's a good way of handling it. I have gone ahead and added enough comments to the function that anyone should be able to figure out how to call it.